core/src/main/scala/kafka/server/AddPartitionsToTxnManager.scala

```diff
@@ -40,13 +40,33 @@ object AddPartitionsToTxnManager {
 
   val VerificationFailureRateMetricName = "VerificationFailureRate"
   val VerificationTimeMsMetricName = "VerificationTimeMs"
+
+  def produceRequestVersionToTransactionSupportedOperation(version: Short): TransactionSupportedOperation = {
+    if (version > 11) {
+      addPartition
+    } else if (version > 10) {
+      genericError
+    } else {
+      defaultError
+    }
+  }
+
+  def txnOffsetCommitRequestVersionToTransactionSupportedOperation(version: Short): TransactionSupportedOperation = {
+    if (version > 4) {
+      addPartition
+    } else if (version > 3) {
+      genericError
+    } else {
+      defaultError
+    }
+  }
 }
 
 /**
  * This is an enum which handles the Partition Response based on the Request Version and the exact operation
  *    defaultError:       This is the default workflow which maps to cases when the Produce Request Version or the Txn_offset_commit request was lower than the first version supporting the new Error Class
  *    genericError:       This maps to the case when the clients are updated to handle the TransactionAbortableException
- *    addPartition:       This is a WIP. To be updated as a part of KIP-890 Part 2
+ *    addPartition:       This allows the partition to be added to the transactions inflight with the Produce and TxnOffsetCommit requests.
```

Comment:
Is this the right code model to represent client capabilities?  The way it's coded it looks like there are 3 mutually exclusive options, but the options are not mutually exclusive: clients that support txnv2 can also support rollback error.